src/test/scala/mailo/PersistenceSpec.scala

```diff
@@ -0,0 +1,174 @@
+}
+
+class PersistenceSpec
+  extends TestKit(ActorSystem("testSystem"))
+  with ImplicitSender
+  with WordSpecLike
+  with Matchers
+  with BeforeAndAfterAll
+  with BeforeAndAfterEach
+  with ScalaFutures {
+  import Eventually._
+
+  implicit def executionContext: ExecutionContext = system.dispatcher
+ 
+  override def afterAll: Unit = {
+    TestKit.shutdownActorSystem(system)
+  }
+
+  override def beforeEach(): Unit = {
+    val tp = TestProbe()
+    tp.send(StorageExtension(system).journalStorage, InMemoryJournalStorage.ClearJournal)
+    tp.send(StorageExtension(system).snapshotStorage, InMemorySnapshotStorage.ClearSnapshots)
+  }
+
+  implicit val defaultPatience =
+      PatienceConfig(timeout = Span(20, Seconds), interval = Span(5, Seconds))
+
+  "persistence actor" should {
+    "properly deliver email messages" in {
+      val state = new ConcurrentLinkedQueue[SimpleMail]()
+      val emailSender = new EmailSender(new MockedData, new MockedClient(state))
+      val loggingActor = system.actorOf(LoggingActor.props())
+      val emailPersistanceActor = system.actorOf(EmailPersistanceActor.props(emailSender, loggingActor))
+
+      val mail1 = Mail(
+        to = "mailo@buildo.io",
+        from = "Mailo mailo@buildo.io",
+        subject = "1",
+        templateName = "mail.html",
+        params = Map.empty,
+        tags = List("test")
+      )
+
+      val mail2 = mail1.copy(subject = "2")
+
+      emailPersistanceActor ! SendEmail(mail1)
+      expectMsg(Queued)
+
+      emailPersistanceActor ! SendEmail(mail2)
+      expectMsg(Queued)
+
+      eventually(timeout(Span(5, Seconds))) {
+        state.size should be(2)
+        info(s"${state.size} messages sent")
+      }
+    }
+
+    "should deliver the correct amount of messages" in {
+      val state = new ConcurrentLinkedQueue[SimpleMail]()
+      val emailSender = new EmailSender(new MockedData, new MockedClientWithDelay(state))
+      val loggingActor = system.actorOf(LoggingActor.props())
+      val emailPersistanceActor = system.actorOf(EmailPersistanceActor.props(emailSender, loggingActor))
+
+      implicit val enqueueTimeout: Timeout = Timeout(100.milliseconds)
+
+      val mail = Mail(
+        to = "mailo@buildo.io",
+        from = "Mailo mailo@buildo.io",
+        subject = "1",
+        templateName = "mail.html",
+        params = Map.empty,
+        tags = List("test")
+      )
+
+      var queuedEmails = 0
+      info("start")
+
+      val task = system.scheduler.schedule(0.seconds, 1.milliseconds,
+        new Runnable {
+          def run(): Unit = {
+            ask(emailPersistanceActor, SendEmail(mail))
+              .onComplete {
+                case scala.util.Success(result) => queuedEmails +=1
+                case scala.util.Failure(error) => ()
+              }
+        }
+      })
+
+      Thread sleep 5000
+
+      emailPersistanceActor ! Kill
+
+      Thread sleep 200
+      task.cancel()
+
```

Comment:
Perhaps it could sleep again after canceling to let the task stop sending email requests, allowing the persistence actor to possibly catch up.

Reply:
👍